### PR TITLE
Fix issues with URI versioning and version annotation condition

### PR DESCRIPTION
**Pull request by supersmile2009**
This PR is based on #1639. Author of the original PR hasn't been responding since 2016.
I rebased commits from the original PR and implemented some changes requested by @GuilhemN.

### Fixed
- Issues with URI versioning (#1529, #1691)
- Version annotation not adding condition for API versioning (#1491)

### CODE REVIEW

**Comment by supersmile2009** [comment]
I didn't completely understand what exactly @GuilhemN meant by these 2 remarks:
https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1639#discussion_r97138151
https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1639#discussion_r97137736
So I'll appreciate if you give more details when you review this PR.

I'm really determined to get this bug properly fixed. We need this feature at our project, so I'm available to add any fixes you request to make this PR accepted.

File: Tests/EventListener/VersionExclusionListenerTest.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            ->disableOriginalConstructor()
+            ->getMock();
+        $event
+            ->expects($this->once())
+            ->method('getRequest')
+            ->willReturn($request);
+
+        $this->listener->onKernelRequest($event);
+
+        $this->assertFalse($request->attributes->has('version'));
```
Comment (GuilhemN) [inline Tests/EventListener/VersionExclusionListenerTest.php]: I don't get what is this test supposed to assert, the new listener does not update the request at all. Shouldn't it be removed?

File: Routing/Loader/Reader/RestActionReader.php [hunk 2]
```diff
@@ -379,6 +391,32 @@ private function combineConditions($conditionOne, $conditionTwo)
...
+     *
+     * @return bool
+     */
+    private function hasVersionPlaceholder($path)
+    {
+        if (false === strpos($path, '{version}')) {
+            return false;
+        }
+
+        return true;
```
Comment (GuilhemN) [inline Routing/Loader/Reader/RestActionReader.php]: `return false !== strpos($path, '{version}');`

**Comment by GuilhemN** [comment]
That's great, thanks a lot for this fix many were awaiting :)

**Pull request merged**